src/unix/linux.c

```diff
@@ -298,6 +568,231 @@ int uv__io_check_fd(uv_loop_t* loop, int fd) {
 }
 
 
+/* Caller must update tail with memory_order_release semantics. */
+static struct uv__io_uring_sqe* uv__iou_get_sqe(struct uv__iou* iou,
+                                                uv_loop_t* loop,
+                                                uv_fs_t* req) {
+  struct uv__io_uring_sqe* sqe;
```

Comment:
Are you planning to support something else in addition to `uv_fs_t` requests? If that's the case maybe have a generic function that takes a void* to fill the `user_data` field is useful, or naming it to reflect that's only for fs requests.

Reply:
Only `uv_fs_t` for now and, since that's encoded in the function prototype, I don't really see the point of repeating it in the function name.

I've worked quite a bit with io_uring over the last year and I'm not that hopeful it's going to be useful for network i/o (or maybe I should say: will outperform epoll.)

To get the most mileage out you need to architect your code around io_uring's way of doing things but that's unfortunately not very compatible with libuv's way of doing things. I'll put together my notes and post them for discussion sometime.